zyte_api/_retry.py

```diff
@@ -182,7 +215,10 @@ class CustomRetryFactory(RetryFactory):
     temporary_download_error_wait = network_error_wait
     throttling_stop = stop_never
     network_error_stop = stop_after_uninterrupted_delay(15 * 60)
-    temporary_download_error_stop = stop_on_count(4)
+    temporary_download_error_stop = stop_on_download_error(max_total=4, max_permanent=2)
```

Comment:
this now handles permanent errors as well, it seems we should change the attribute name. It'd make the changes backwards incompatible, but I think that's fine, as soon as it's documented.